pandas/core/arrays/datetimelike.py

```diff
@@ -1138,8 +1177,8 @@ def _time_shift(self, periods, freq=None):
                 freq = frequencies.to_offset(freq)
             offset = periods * freq
             result = self + offset
-            if hasattr(self, 'tz'):
-                result._tz = self.tz
+            if getattr(self, 'tz', None):
+                result._dtype = DatetimeTZDtype(tz=self.tz)
```

Comment:
simpler just to set result._dtype = self._dtype?

Reply:
I think this is unnecessary now. `result.dtype` is already correct.


```python
(Pdb) l
1176                if isinstance(freq, compat.string_types):
1177                    freq = frequencies.to_offset(freq)
1178                offset = periods * freq
1179                result = self + offset
1180                import pdb; pdb.set_trace()
1181 ->             if getattr(self, 'tz', None):
1182                    result._dtype = DatetimeTZDtype(tz=self.tz)
1183                return result
1184
1185            if periods == 0:
1186                # immutable so OK
(Pdb) self.dtype
datetime64[ns, UTC]
(Pdb) result.dtype
datetime64[ns, UTC]
```